### DISCUSSION TITLE
Island with a single word in filename and kebab-case?

### DISCUSSION THREAD

**Discussion by neezer** [freshframework/fresh#627] [Q&A] [answered]
I made an island initially named `Select.tsx`, then I changed my mind and wanted it named `select.tsx`, so I renamed it. Fresh compiles and runs fine with this change, but VSCode has been complaining ever since:

> File name 'file:///path/to/islands/select.tsx' differs from already included file name 'file:///path/to/islands/Select.tsx' only in casing.
>   The file is in the program because:
>     Root file specified for compilation    Imported via "./select.tsx" from file 'file:///path/to/islands/select-department.tsx'deno-ts(1149)

The [Fresh docs](https://fresh.deno.dev/docs/getting-started/adding-interactivity) say:

> The name of the module should be the [pascal case](https://en.wiktionary.org/wiki/Pascal_case) or [kebab case](https://en.wiktionary.org/wiki/kebab_case) name of the island component.

A single-word kebab-case would just be `select.tsx`, but I'm wondering if this is something like the lack of a capital `S` and the lack of a hyphen `-` is causing issues? I feel like this is a Fresh (cache?) issue, as I've used filenames like `select.tsx` in other Deno projects that aren't powered by Fresh and VSCode is just dandy with those. I also have another island named `select-department.tsx` and VSCode has **_no_** warning for that import.

Anyone have any insight here? I'd really prefer to use kebab-case and not have PascalCase files littered in only because they're only one word long... 🤔

**Comment by marvinhagemeister** [comment, marked as answer]
This is not an issue with Fresh but with file system case sensitivity. By default, Mac OS uses a case-insensitive file system to support compatibility and this trips up some tools when only changing casing when renaming a file. There is a long running issue (and probably others) in the vscode issue tracker https://github.com/microsoft/vscode/issues/94307
